docs/cudf/source/developer_guide/benchmarking.md

```diff
@@ -0,0 +1,288 @@
+they should also assume as little as possible about the nature of the data.
+For instance, unless there are meaningful functional differences,
+benchmarks should not care about the dtype or nullability of the data.
+Objects that differ in these ways should be interchangeable for most benchmarks.
+The goal of writing benchmarks in this way is to then automatically benchmark objects with different properties.
+We support this use case with the `benchmark_with_object` decorator.
+
+The use of this decorator is best demonstrated by example:
+
+```python
+@benchmark_with_object(cls="dataframe", dtype="int", cols=6)
+def bench_foo(benchmark, dataframe):
+    benchmark(dataframe.foo)
+```
+
+In the example above `bench_foo` will be run for DataFrames containing six columns of integer data.
+The decorator allows automatically parametrizing the following object properties:
+
+- `cls`: Objects of a specific class, e.g. `DataFrame`.
+- `dtype`: Objects of a specific dtype.
+- `nulls`: Objects with and without null entries.
+- `cols`: Objects with a certain number of columns.
+- `rows`: Objects with a certain number of rows.
+
+In the example, since we did not specify the number of rows or nullability,
+it will be run once for each valid number of rows and for both nullable and non-nullable data.
+The valid set of all parameters (e.g. the numbers of rows) is stored in the `common/config.py` file.
+This decorator allows a developer to write a generic benchmark that works for many types of objects,
+then have that benchmark automatically run for all objects of interest.
+
+### Custom fixtures
+
+Developers may define custom fixtures if necessary, but this should be done with care.
+The `benchmark_with_object` decorator covers most use cases and automatically guarantees a baseline of benchmark coverage.
+When writing fixtures, developers should make the data sizes dependent on the benchmarks configuration.
+The `benchmarks/common/config.py` file defines standard data sizes to be used in benchmarks.
+These data sizes can be tweaked for debugging purposes (see {ref}`testing` below).
+Fixture sizes should be relative to the `NUM_ROWS` and/or `NUM_COLS` variables defined in the config module.
+
+### Parametrization vs. fixtures
+
+Our benchmarks make use of the [`pytest_cases`](https://smarie.github.io/python-pytest-cases/) `pytest` plugin.
+This plugin allows us to handle parametrization much more cleanly than pytest does out of the box.
+Specifically, it provides some syntactic sugar around
+
+```python
+@pytest.mark.parametrize(
+    "num", [1, 2, 3]
+)
+def bench_foo(benchmark, num):
+    benchmark(num * 2)
+```
+
+for when the parameters are nontrivial and require complex initialization.
+This is common for benchmarks of functions accepting cuDF objects, such as `cudf.concat`.
+With `pytest_cases`, the different cases are instead placed into separate functions and automatically made available.
```

Comment:
Since this is specifically referring to a package, should we leave a link for further reading?

```suggestion
With [`pytest_cases`](https://smarie.github.io/python-pytest-cases/), the different cases are instead placed into separate functions and automatically made available.
```